### PR TITLE
Add sensu logging to /var/log/sensu/ on systemd based systems.

### LINKED ISSUE #1292: Document configuration for writing service logs to file on systems using journald
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

## Expected Behavior

As an operator I can learn about configuring my Sensu installation to write logs to files on disk.

## Current Behavior

Operators must do background research on journald, syslog and logrotate.

## Possible Solution

Document process for sending logs from journald to rsyslog, and configuring rsyslog to write Sensu service logs to files in /var/log/sensu

## Context

The desire to log to a file is becoming a somewhat frequent request from a variety of users.

### PR DESCRIPTION
**Pull request by rgeniesse**
## Description
Closes #1292 

## Motivation and Context
Give users the option to have log files written to `/var/log/sensu/servicename.log` when `systemd` is the service manager. While `journalctl` does give you access to what you need, it's a nice option to have a more traditional logging system in place.

### CODE REVIEW

File: content/sensu-go/5.5/guides/systemd-logs.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+title: "Sensu service logging with systemd"
+description: "place holder"
+weight: 2000
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/systemd-logs.md]:
```suggestion
```

File: content/sensu-go/5.5/guides/systemd-logs.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+version: "5.5"
+product: "Sensu Go"
+platformContent: false
+---
+
+## Adding logging to disk for systemd
+
+By default, systems where `systemd` is the service manager do not write logs to `/var/log/sensu/` for the `sensu-agent` and the `sensu-backend` services. The following will walk you how to add log forwarding from `journald` to `syslog`, have `rsyslog` write logging data to disk and setup log rotation of the newly created log files.
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/systemd-logs.md]:
```suggestion
By default, systems where `systemd` is the service manager do not write logs to `/var/log/sensu/` for the `sensu-agent` and the `sensu-backend` services. This guide walks you through how to add log forwarding from `journald` to `syslog`, have `rsyslog` write logging data to disk, and set up log rotation of the newly created log files.
```

File: content/sensu-go/5.5/guides/systemd-logs.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+To configure `journald` to forward logging data to `syslog`, modify `/etc/systemd/journald.conf` to include the following line:
+
+{{< highlight shell >}}
+ForwardToSyslog=yes
+{{< /highlight >}}
+
+Next, setup `rsyslog` to write the logging data received from `journald` to `/var/log/sensu/servicename.log`. In this example, the `sensu-backend` and `sensu-agent` logging data is sent to individual files named after the sercice. The `sensu-backend` is not required if only setting up log forwarding for the `sensu-agent` service:
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/systemd-logs.md]:
```suggestion
Next, set up `rsyslog` to write the logging data received from `journald` to `/var/log/sensu/servicename.log`. In this example, the `sensu-backend` and `sensu-agent` logging data is sent to individual files named after the service. The `sensu-backend` is not required if only setting up log forwarding for the `sensu-agent` service.
```

File: content/sensu-go/5.5/guides/systemd-logs.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+---
+title: "Sensu service logging with systemd"
+description: "place holder"
+weight: 2000
+version: "5.5"
+product: "Sensu Go"
+platformContent: false
+---
+
+## Adding logging to disk for systemd
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/systemd-logs.md]:
```suggestion
```

File: content/sensu-go/5.5/guides/systemd-logs.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+{{< /highlight >}}
+
+Restart `rsyslog` and `journald` to apply the new configuration:
+
+{{< highlight shell>}}
+systemctl restart systemd-journald
+systemctl restart rsyslog
+{{< /highlight>}}
+
+Setup log rotation for newly created log files. This example will rotate all log files in `/var/log/sensu/` weekly, unless the size of 100M is reached first. The last five rotated logs will be kept and they will be compressed, with the exception of the most recent one.
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/systemd-logs.md]:
```suggestion
Set up log rotation for newly created log files. This example rotates all log files in `/var/log/sensu/` weekly, unless the size of 100M is reached first. The last five rotated logs are kept and compressed, with the exception of the most recent one.
```

File: content/sensu-go/5.5/guides/troubleshooting.md [hunk 6]
```diff
@@ -47,10 +47,10 @@ consult the Operating section of the [agent][agent-ref] or
 
 Sensu services print [structured log messages][structured] to standard output.
 In order to capture these log messages to disk or another logging facility, Sensu services
-make use of capabilities provided by the underlying operating service's service
+make use of capabilities provided by the underlying operating system's service
 management. For example, logs are sent to the journald when systemd is the service manager,
 whereas log messages are redirected to `/var/log/sensu` when running under sysv
-init schemes.
+init schemes. If you are running systemd as your service manager and would rather have logs written to `/var/log/sensu/`, you can follow our guide on forwarding logs from [journald to syslog][journald-syslog].
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/troubleshooting.md]:
```suggestion
init schemes. If you are running systemd as your service manager and would rather have logs written to `/var/log/sensu/`, see the guide to [forwarding logs from journald to syslog][journald-syslog].
```

**Review by apaskulin — COMMENTED** [review]
Nice work! I ran through this on CentOS 7, and it worked great. I'd recommend switching terms like syslog, systemd, journald, etc to regular styling instead of code styling since there's already a lot of in-line code styling in the guide.

File: content/sensu-go/5.5/guides/systemd-logs.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
+---
+title: "Sensu service logging with systemd"
+description: "place holder"
```
Comment (apaskulin) [inline content/sensu-go/5.5/guides/systemd-logs.md]:
```suggestion
description: "By default, systems where systemd is the service manager do not write logs to /var/log/sensu/. This guide walks you through how to add log forwarding from journald to syslog, have rsyslog write logging data to disk, and set up log rotation of the newly created log files."
```

**Review by apaskulin — APPROVED** [review]
Just one suggestion then +1 to copy to all versions

68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f643546624d31357642323744465a433254502f67697068792e676966

**Comment by asachs01** [comment]
@apaskulin @rgeniesse was it intended that this show up under guides?

**Comment by rgeniesse** [comment]
@asachs01 it was not. The guides list is pretty long already. The discoverability of it is through the troubleshooting document or searching.

**Pull request merged**